### PR TITLE
context_extraction win32 patch

### PR DESCRIPTION
**Pull request by YellowOnion**
Untested as full program doesn't compile, but would hate for this to go to waste.

### CODE REVIEW

File: libopenage/gui/guisys/private/platforms/context_extraction_win32.cpp [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+namespace qtsdl {
+
+std::tuple<QVariant, WId> extract_native_context(SDL_Window *window) {
+	assert(window);
+
+	HGLRC current_context;
+	SDL_SysWMinfo wm_info;
+	SDL_VERSION(&wm_info.version);
+	if (SDL_GetWindowWMInfo(window, &wm_info)) {
+#ifndef NDEBUGwglGetCurrentContext
```
Comment (ChipmunkV) [inline libopenage/gui/guisys/private/platforms/context_extraction_win32.cpp]: That ifndef is probably some kind of leftover.

**Comment by ChipmunkV** [comment]
That's fine, I've tested a standalone variation in here: [QTBUG-50309](https://bugreports.qt.io/browse/QTBUG-50309). You can stick the code of this PR there and check (if you have time and want to be 100% sure).

To fix validation errors, add your name to `copying.md` and probably change the year to 2016-2016 in `context_extraction_win32.cpp`.

File: libopenage/gui/guisys/private/platforms/context_extraction_win32.cpp [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+#endif
+		current_context = wglGetCurrentContext();
+		assert(current_context);
+	}
+    QWGLNativeContext nativeContext(current_context, wm_info.info.win.window);
```
Comment (TheJJ) [inline libopenage/gui/guisys/private/platforms/context_extraction_win32.cpp]: indent this line with a tab pls :)

File: libopenage/gui/guisys/private/platforms/context_extraction_win32.cpp [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+	return std::make_tuple(QVariant::fromValue(nativeContext), (WId)wm_info.info.win.window);
```
Comment (TheJJ) [inline libopenage/gui/guisys/private/platforms/context_extraction_win32.cpp]: can you use a c++ style cast for `(WId)` please? Probably `static_cast` wil do.

Comment (ChipmunkV) [inline libopenage/gui/guisys/private/platforms/context_extraction_win32.cpp]: Yes, it's `void*` to `int` - `reinterpret_cast`.

**Comment by ChipmunkV** [comment]
Btw, can you add a `guisys:` prefix to the commit message, pls?

I don't know if it'll be useful in the future, but it's kind of recommended in `doc/code/gui.md`.

**Comment by TheJJ** [comment]
Merged with #867

**Pull request closed without merging**